source/os/source/threads.cpp

```diff
@@ -39,7 +39,8 @@ bool Mutex::try_lock() {
 	int id = getThreadID();
 	if(owningThread != id) {
 		for(unsigned i = 0; i < spinCount; ++i) {
-			if(owningThread.compare_exchange_strong(id, invalidThreadID) == invalidThreadID) {
```

Comment:
the std::atomic version of compare_exchange_strong has the arguments switched, and returns bool instead of the value currently being stored. 

The first argument is also a reference, and has it's value replaced with the value stored in the atomic object at the time the operation is initiated. Previously this would have been the return value.